### PR TITLE
JP-4318: Updating Ramp Fitting Tests for New STCAL Slope Computation.

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
Resolves [JP-4318](https://jira.stsci.edu/browse/JP-4318)

This PR addresses using the median rate of a pixel, which is computed across all integrations of a pixel, for the computation of the rate and rateint slopes, using the Poisson variance as part of the weights. This should not happen, since this has the effect of later integrations affecting the slope of earlier integrations. To prevent this the weighting during the computation of rate and rateint slopes use only the read noise variance, rather than a combination of both Poisson and read noise variances.

This is linked with the STCAL PR https://github.com/spacetelescope/stcal/pull/550

## Tasks
- [ ] If you have a specific reviewer in mind, tag them.
- [ ] add a build milestone, i.e. `Build 12.0` (use the [latest build](https://github.com/spacetelescope/jwst/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see [changelog readme](https://github.com/spacetelescope/jwst/blob/main/changes/README.rst) for instructions) 
    - if your change breaks **step-level or public API** ([as defined in the docs](https://jwst.readthedocs.io/en/latest/jwst/user_documentation/more_information.html#api-public-vs-private)), also add a `changes/<PR#>.breaking.rst` news fragment
  - [ ] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [ ] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by kmacdonald-stsci** [comment]
Regression test run:

https://github.com/spacetelescope/RegressionTests/actions/runs/26763461997

File: jwst/ramp_fitting/tests/test_ramp_fit.py [hunk 1]
```diff
@@ -380,6 +380,7 @@ def test_four_groups_two_CRs_at_end(self, method):
         np.testing.assert_allclose(slopes["slope"][50, 50], cds_slope, 1e-6)
 
     def test_four_groups_four_CRs(self, method):
+        """Weird edge case with slope zero."""
```
Comment (tapastro) [inline jwst/ramp_fitting/tests/test_ramp_fit.py:383]: I'm not sure I understand this scenario. If a pixel has the jump flag for every measurement, do we assign it a rate of 0.0? I would have imagined that would not result in a rate, leading to NaN + dq->DO_NOT_USE. Why did the weight change affect this unit test?

Any thoughts @drlaw1558 ?

Comment (drlaw1558) [reply jwst/ramp_fitting/tests/test_ramp_fit.py:383]: I'm not certain, but my *guess* would be that its some special case in which both var_p=0 and var_r=0 because there are no groups to fit to.  Weights are 1/variance and maybe we're catching the special case where var_r=0 to avoid divide-by-zero errors but weren't catching this when we were using var_e instead.

The interesting question in my mind is whether this means that we'll get zeros for bad pixels in places where we used to get NaN.  I tested this just now by hacking a NIRSpec file to force this same case of JUMP_DET set for all pixels in a ramp and it *does* look like this is happening.  In a pixel I set to JUMP_DET everywhere it used to have NaN valued flux, ERR=0 and DQ=JUMP_DET, while now it has flux=0, error=0, and DQ=JUMP_DET.

We may need a special catch for the case where there are no non-zero weights when combining segments, and set SCI=NaN, ERR=NaN, DQ=DO_NOT_USE.

Comment (kmacdonald-stsci) [reply jwst/ramp_fitting/tests/test_ramp_fit.py:383]: Is this even a valid edge case? Is it possible for all groups to be flagged as JUMP_DET?  Is this just a test created to create edge cases someone could dream up or is it something that could actually happen that we should worry about? Every time I see this test I wonder if it's worth keeping.

Comment (drlaw1558) [reply jwst/ramp_fitting/tests/test_ramp_fit.py:383]: Unclear.  I pulled up some 2-group NIRCam data as the most likely case in which all groups might be flagged, and it looks like JUMP_DET isn't set anywhere (and there's no different in rates before/after this PR), which makes me think that it can't actually flag all groups.  It's certainly an extreme edge case at best.

Comment (tapastro) [reply jwst/ramp_fitting/tests/test_ramp_fit.py:383]: It does appear, reading the summary of the algorithm here: https://stcal.readthedocs.io/en/latest/stcal/jump/description.html#two-point-difference-method

that jump does not allow itself to flag all segments - assuming of course that our documentation matches code behavior.

I don't think this deserves more commits here, but it does expose the test as covering a case I don't think we can generate. Maybe best to move forward with this work and open a follow up ticket to investigate whether we can ever generate a ramp with all segments flagged as jump?

**Review by tapastro — APPROVED** [review]
